.github/workflows/test-r-scripts.yaml

```diff
@@ -1,4 +1,12 @@
-on: [push, pull_request]
+on:
+  push:
+    branches:
+      - main
+      - master
+  pull_request:
+    branches:
+      - main
+      - master
```

Comment:
As for the workflow to run R CMD check, also here we are more intentional about what branches trigger gh-actions.